File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    
    int tmp = 0, zero_nums = 0;
    for(int i=0; i<lottos.size(); i++){
        if(lottos[i] == 0){
            zero_nums++;
            continue;
        }
        
        // 디폴트 갯수
        if(find(win_nums.begin(), win_nums.end(), lottos[i]) != win_nums.end())
            tmp++;
    }
    
    if(tmp+zero_nums < 1)
        answer.push_back(6);
    else
        answer.push_back(7-(tmp+zero_nums));
    
    if(tmp < 1)
        answer.push_back(6);
    else
        answer.push_back(7-tmp);
    
    return answer;
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.js
function solution(lottos, win_nums) {
    var answer = [];
    
    let correct = lottos.filter((lotto) => win_nums.includes(lotto)).length;
    
    let zero_nums = lottos.filter((lotto) => lotto === 0).length;
    
    let min = 7-correct >= 6 ? 6 : 7-correct;
    let max = min-zero_nums < 1 ? 1 : min-zero_nums;
    
    answer.push(max);
    answer.push(min);
    
    return answer;
}